File: Cryptography/Haskell/caesar_cipher.hs
import IO
import Char

caesar_encrypt_letter :: Int -> Char -> Char
caesar_encrypt_letter key letter =
	if isAlpha(letter)
		then chr(((ord(toUpper(letter)) - ord('A') + key) `mod` 26) + ord('A'))
		else letter


caesar_decrypt_letter :: Int -> Char -> Char
caesar_decrypt_letter key letter =
	if isAlpha(letter)
		then chr(((ord(toUpper(letter)) - ord('A') - key) `mod` 26) + ord('A'))
		else letter


caesar_encrypt :: String -> Int -> String
caesar_encrypt plaintext key = 
	map (caesar_encrypt_letter key) plaintext 


caesar_decrypt :: String -> Int -> String
caesar_decrypt ciphertext key = 
	map (caesar_decrypt_letter key) ciphertext 


caesar_brute_force :: String -> IO()
caesar_brute_force ciphertext =
	mapM_ (\x -> putStrLn ((show x) ++ "\t" ++ caesar_decrypt ciphertext x)) [1 .. 25]


main = do
	let plaintext = "Meet me after the toga party"
	let ciphertext = (caesar_encrypt plaintext 3)
	putStrLn ("Plaintext: " ++ plaintext)
	putStrLn ("Ciphertext: " ++ ciphertext)

	putStrLn ("\nBrute forcing Ciphertext " ++ ciphertext)
	
	caesar_brute_force (caesar_encrypt "Meet me after the toga party" 3)

File: Cryptography/python/caesar_cipher/caesar_cipher.py
#!/usr/bin/python

class CaesarCipher:
	def __init__(self, key):
		self.key = key

	def encrypt(self, plaintext):
		ciphertext = ""
		for p in plaintext:
			if (p.isalpha()):
				ciphertext += (chr (((ord(p.upper()) - ord('A') + self.key) % 26) + ord('A'))).upper()
			else:
				ciphertext += p

		return ciphertext

	def decrypt(self, ciphertext):
		plaintext = ""
		for c in ciphertext:
			if (c.isalpha()):
				plaintext += (chr (((ord(c.upper()) - ord('A') - self.key) % 26) + ord('A'))).lower()
			else:
				plaintext += c
	
		return plaintext

	@staticmethod
	def brute_force(ciphertext):
		print "Brute forcing Ciphertext", ciphertext
		for i in range(1, 26):
			print i, "\t", CaesarCipher(i).decrypt(ciphertext)

def main():
	cipher = CaesarCipher(3)
	t = "Meet me after the toga party"
	c = cipher.encrypt(t)
	p = cipher.decrypt(c)

	print 'Plaintext:', t
	print 'Ciphertext:', c
	print

	CaesarCipher.brute_force(c)

if __name__ == "__main__":
	main()
